File: backend/models/workout_plan.py
from typing import List
from pydantic import BaseModel, Field

from backend.models.exercise import PlannedExercise
from backend.models.user_profile import UserProfile


class PlanMetadata(BaseModel):
    plan_name: str
    plan_type: str
    difficulty: str
    estimated_session_time_minutes: int


class WorkoutDay(BaseModel):
    day_number: int
    day_name: str
    focus: str
    exercises: List[PlannedExercise] = Field(default_factory=list)


class WorkoutPlan(BaseModel):
    user_profile: UserProfile
    plan_metadata: PlanMetadata
    days: List[WorkoutDay] = Field(default_factory=list)
    days_requested: int = 0
    days_generated: int = 0
    warnings: List[str] = Field(default_factory=list)

File: frontend/src/types/workoutPlan.ts
export interface PlannedExercise {
  name: string;
  sets: number;
  reps: string;
  rest_seconds: number;
}

export interface WorkoutDay {
  day_number: number;
  day_name: string;
  focus: string;
  exercises: PlannedExercise[];
}

export interface PlanMetadata {
  plan_name: string;
  plan_type: string;
  difficulty: string;
  estimated_session_time_minutes: number;
}

export interface UserProfile {
  sex: string;
  age: number;
  height_cm: number;
  weight_kg: number;
  experience_level: string;
  days_per_week: number;
  available_equipment: string[];
}

export interface WorkoutPlan {
  user_profile: UserProfile;
  plan_metadata: PlanMetadata;
  days: WorkoutDay[];
  days_requested: number;
  days_generated: number;
  warnings: string[];
}

export interface StoredWorkoutPlan extends WorkoutPlan {
  id: string;
  created_at: string;
}
